### PR TITLE
CH4 Cancel Cleanup

### PR DESCRIPTION
**Pull request by wesbland**
Locks were being unnecessary cycled in the CH4 netmod/shmemod layers. This moves all of that up to the CH4 layer so things are consistent. The original authors are @charlesarcher and @zhenggb72.

The changes in #2369 sit on top of this PR.

### CODE REVIEW

**Comment by wesbland** [comment]
The more contentious parts of this patch were merged a while back. I think it's probably simple enough that it can be merged now, but it'd be good to have another opinion on that, @raffenet?

**Comment by wesbland** [comment]
test:jenkins/ch4/ofi

**Comment by wesbland** [comment]
test:jenkins/ch4/ofi

**Comment by wesbland** [comment]
The tests still aren't all passing here, but this patch does fix all of the `MPI_ANY_SOURCE` failures in the debug builds due to unfreed handles.

File: src/mpid/ch4/netmod/ofi/ofi_recv.h [hunk 1]
```diff
@@ -42,8 +42,11 @@ MPL_STATIC_INLINE_PREFIX int MPIDI_OFI_do_irecv(void *buf,
     MPIR_FUNC_VERBOSE_STATE_DECL(MPID_STATE_MPIDI_OFI_DO_IRECV);
     MPIR_FUNC_VERBOSE_ENTER(MPID_STATE_MPIDI_OFI_DO_IRECV);
 
-    if (mode == MPIDI_OFI_ON_HEAP)      /* Branch should compile out */
+    if (mode == MPIDI_OFI_ON_HEAP) {        /* Branch should compile out */
```
Comment (raffenet) [inline src/mpid/ch4/netmod/ofi/ofi_recv.h]: extra whitespace here

**Comment by wesbland** [comment]
Fixed your whitespace comment. This should be ready for your signoff @raffenet.

File: src/mpid/ch4/netmod/ofi/ofi_recv.h [hunk 2]
```diff
@@ -274,20 +274,13 @@ MPL_STATIC_INLINE_PREFIX int MPIDI_NM_mpi_cancel_recv(MPIR_Request * rreq)
     MPIR_FUNC_VERBOSE_STATE_DECL(MPID_STATE_MPIDI_NM_MPI_CANCEL_RECV);
     MPIR_FUNC_VERBOSE_ENTER(MPID_STATE_MPIDI_NM_MPI_CANCEL_RECV);
 
-    if (!MPIDI_OFI_ENABLE_TAGGED) {
```
Comment (raffenet) [inline src/mpid/ch4/netmod/ofi/ofi_recv.h]: I don't think this should be removed? Maybe it was a bad conflict resolution...

Comment (wesbland) [inline src/mpid/ch4/netmod/ofi/ofi_recv.h]: Hmm. Good question. Well @charlesarcher isn't around anymore for us to ask, but I think you're right. I'll revert this part out.

Comment (wesbland) [inline src/mpid/ch4/netmod/ofi/ofi_recv.h]: Done.

**Pull request merged**